### PR TITLE
F add storage containers

### PR DESCRIPTION
**Pull request by mbfrahry**
This PR adds `storage_containers` to the Provider. It also has all resources use the `OPCClient` which holds a compute client and a storage client for their respective resources. Java and Database clients will also be added to `OPCClient` at a future date. To account for the `OPCClient` shift, all resources had to be refactored to use `OPCClient.computeClient`

```
TF_ACC=1 make test TEST=./opc TESTARGS="-v -run=TestAccOPCStorageContainer"
==> Checking that code complies with gofmt requirements...
go test -i ./opc || exit 1
echo ./opc | \
		xargs -t -n4 go test -v -run=TestAccOPCStorageContainer -timeout=300s -parallel=4
go test -v -run=TestAccOPCStorageContainer -timeout=300s -parallel=4 ./opc
=== RUN   TestAccOPCStorageContainer_importBasic
--- PASS: TestAccOPCStorageContainer_importBasic (14.48s)
=== RUN   TestAccOPCStorageContainer_Basic
--- PASS: TestAccOPCStorageContainer_Basic (13.18s)
=== RUN   TestAccOPCStorageContainer_Updated
--- PASS: TestAccOPCStorageContainer_Updated (25.87s)
PASS
ok  	github.com/terraform-providers/terraform-provider-opc/opc	53.564s
```

### CODE REVIEW

File: opc/config.go [hunk 1]
```diff
@@ -58,7 +60,24 @@ func (c *Config) Client() (*compute.ComputeClient, error) {
...
+	computeClient, err := compute.NewComputeClient(&config)
+	if err != nil {
+		return nil, err
+	}
+
+	opcClient := &OPCClient{
+		computeClient: computeClient,
+	}
+
+	if c.Storage {
```
Comment (grubernaut) [inline opc/config.go]: hm, I guess we can't initialize both of the clients at the same time, as some users may not have access to the storage features?

Comment (grubernaut) [inline opc/config.go]: Would much rather have this without the extra boolean configuration, but I guess it's necessary?

File: opc/provider_test.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ func TestProvider_impl(t *testing.T) {
 }
 
 func testAccPreCheck(t *testing.T) {
-	required := []string{"OPC_USERNAME", "OPC_PASSWORD", "OPC_IDENTITY_DOMAIN", "OPC_ENDPOINT"}
+	required := []string{"OPC_USERNAME", "OPC_PASSWORD", "OPC_IDENTITY_DOMAIN", "OPC_ENDPOINT", "OPC_STORAGE"}
```
Comment (grubernaut) [inline opc/provider_test.go]: Feels like this env var should be more clear about what it's expecting as a value

File: opc/provider.go [hunk 3]
```diff
@@ -49,6 +49,13 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("OPC_INSECURE", false),
 				Description: "Skip TLS Verification for self-signed certificates. Should only be used if absolutely required.",
 			},
+
+			"storage": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OPC_STORAGE", false),
```
Comment (grubernaut) [inline opc/provider.go]: This env var should likely be more clear. `EnvDefaultFunc` also returns a string, not a boolean. Should we use a non-default function to allow any non-nil value to enable authentication? 
ie: 
```
DefaultFunc: func() (interface{}, error) {
  if v := os.Getenv(k); v != "" {
    return true, nil
  }
  return false, nil
}
```
Thoughts?

File: opc/resource_storage_container.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"read_acls": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
```
Comment (grubernaut) [inline opc/resource_storage_container.go]: Does this have any ordering issues, as the parent type is a list? ie: does the API return the same items, but in a different order?

Comment (grubernaut) [inline opc/resource_storage_container.go]: Same with the other `TypeList` attributes

File: opc/resource_storage_container_test.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+	return nil
+}
+
+const testAccOPCStorageContainerBasic = `
+resource "opc_storage_container" "test" {
+	name = "acc-storage-container-%d"
+  max_age = 50
+  primary_key = "test-key"
+	allowed_origins = ["origin-1"]
```
Comment (grubernaut) [inline opc/resource_storage_container_test.go]: Odd spacing here

File: opc/resource_storage_container_test.go [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckStorageContainerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckStorageContainerExists,
+					resource.TestCheckResourceAttr(containerResourceName, "max_age", "50"),
+					resource.TestCheckResourceAttr(containerResourceName, "primary_key", "test-key"),
+				),
```
Comment (grubernaut) [inline opc/resource_storage_container_test.go]: Should check for the length of the `allowed_origins` list

File: opc/resource_storage_container_test.go [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+			},
+		},
+	})
+}
+
+func TestAccOPCStorageContainer_Updated(t *testing.T) {
+	containerResourceName := "opc_storage_container.test"
+	ri := acctest.RandInt()
+	config := fmt.Sprintf(testAccOPCStorageContainerBasic, ri)
+	config2 := fmt.Sprintf(testAccOPCStorageContainerUpdated, ri)
```
Comment (grubernaut) [inline opc/resource_storage_container_test.go]: Possible to test appending to the list as well? `["origin2", "origin3"]`?

File: opc/config.go (hunk 1 above)
Comment (mbfrahry) [inline opc/config.go]: Do you think it'd be better to have them pass in the actual storage url instead?

File: opc/resource_storage_container.go (hunk 4 above)
Comment (mbfrahry) [inline opc/resource_storage_container.go]: It looks like we get it back in the same order we gave it.

File: opc/resource_storage_container_test.go (hunk 6 above)
Comment (mbfrahry) [inline opc/resource_storage_container_test.go]: Done

File: opc/resource_storage_container_test.go (hunk 7 above)
Comment (mbfrahry) [inline opc/resource_storage_container_test.go]: Done

**Pull request merged**